caddyfile.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formatencoder
+
+import (
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+)
+
+// UnmarshalCaddyfile sets up the module from Caddyfile tokens. Syntax:
+//
+//     formatted [<template>] [{
+//			placeholder	[<placeholder>]
+//		}]
+//
+// If the value of "template" is omitted, Common Log Format is assumed.
+// See the godoc on the LogEncoderConfig type for the syntax of
+// subdirectives that are common to most/all encoders.
+func (se *FormattedEncoder) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	for d.Next() {
+		args := d.RemainingArgs()
+		switch len(args) {
+		case 0:
+			se.Template = commonLogFormat
+		case 1:
+			se.Template = args[0]
+		default:
```

Comment:
Alternatively you could just join all the remaining args with spaces, would avoid requiring wrapping with quotes here. It's how CEL expressions are handled usually, works ok in simple cases

Reply:
Done.